common/config_schema.go

```diff
@@ -41,9 +41,21 @@ type RecordKey = struct {
 // Valid objects require one of the following fields to be specified.
 type RequiredFields = []SchemaKey
 
+// for sid and platforms data_type
+type FilterType = struct {
+	// whitelist and blacklist are mutually exclusive
+	// for platforms, sid platforms, string chars
+	Whitelist []string `json:"whitelist,omitempty" msgpack:"whitelist,omitempty"`
+	Blacklist []string `json:"blacklist,omitempty" msgpack:"blacklist,omitempty"`
```

Comment:
```suggestion
	ValidRE []string `json:"valid_re,omitempty" msgpack:"valid_re,omitempty"`
	InvalidRE []string `json:"invalid_re,omitempty" msgpack:"invalid_re,omitempty"`
```
I suggest:
-  Making those fields regular expressions as this will grant people a lot more flexibility (and reduce the likeliness we need to expand these in the future).
- Renaming and include "re" in the name to make it more obvious on the type of validation/language is expected as values.

Reply:
all makes sense, let me just make this work on the front-end before updating this PR